### PR TITLE
[`Masks`] Fix mask handling in eager for vision models

### PR DESCRIPTION
**Pull request by vasqu**
As per title, some vision models use masks, let's sync with bert this time and reduce errors that could be introduced

### CODE REVIEW

**Comment by ydshieh** [comment]
il faut run-slow svp, Monsieur, au moins une parti 🙏

**Comment by vasqu** [comment]
run-slow: audio_spectrogram_transformer, deit, dinov2, dinov2_with_registers, dinov3_vit, dpt, ijepa, videomae, vit, vit_mae, vit_msn, vitpose_backbone, vivit, yolos

**Comment by vasqu** [comment]
Checked against main and the failing tests are the same, videomae in general seems to be broken tho (integration tests)

File: src/transformers/models/audio_spectrogram_transformer/modeling_audio_spectrogram_transformer.py [hunk 1]
```diff
@@ -96,25 +96,28 @@ def forward(self, input_values: torch.Tensor) -> torch.Tensor:
...
 
-    # Normalize the attention scores to probabilities.
-    attn_weights = nn.functional.softmax(attn_weights, dim=-1, dtype=torch.float32).to(query.dtype)
+    if attention_mask is not None:
+        attention_mask = attention_mask[:, :, :, : key.shape[-2]]
+        attn_weights = attn_weights + attention_mask
 
-    # This is actually dropping out entire tokens to attend to, which might
-    # seem a bit unusual, but is taken from the original Transformer paper.
+    attn_weights = nn.functional.softmax(attn_weights, dim=-1)
```
Comment (vasqu) [inline src/transformers/models/audio_spectrogram_transformer/modeling_audio_spectrogram_transformer.py:120]: Only real diff is here where we don't cast in fp32, however this was originally also not the case and was only introduced in the attn refactor (which shouldn't have happened), see https://github.com/huggingface/transformers/pull/36545/files#diff-e3964696c93e4ad2f40a0ffd9f1079e8fc58645d328e8f7ad27f9f9c5dba47b5L225-L226

**Review by NielsRogge — APPROVED** [review]
Thanks! I hope the slow tests pass

**Review by ArthurZucker — APPROVED** [review]
